src/RelationalBehavior.php

```diff
@@ -113,40 +126,71 @@ public function saveRelations($event)
     {
         /** @var ActiveRecord $model */
         $model = $event->sender;
+        $position = 0;
 
         $relatedRecords = $model->getRelatedRecords();
         foreach ($relatedRecords as $relationName => $relationRecords) {
             $activeQuery = $model->getRelation($relationName);
             if (!empty($activeQuery->via)) { // works only for many-to-many relation
                 list($junctionTable, $primaryModelColumn, $relatedModelColumn) = $this->parseQuery($activeQuery);
-
                 $junctionRows = [];
                 $relationPks = ArrayHelper::getColumn($relationRecords, array_keys($activeQuery->link)[0], false);
                 $passedRecords = count($relationPks);
-                $relationPks = array_filter($relationPks);
+
+                if ($this->sortSettings[$relationName] &&
+                    isset($this->sortSettings[$relationName]['sortColumn']) &&
+                    isset($this->relationalRawData[$relationName])
+                ) {
+                    $relationPks = $this->relationalRawData[$relationName];
+                } else {
+                    $relationPks = array_filter($relationPks);
+                }
+
                 $savedRecords = count($relationPks);
                 if ($passedRecords != $savedRecords) {
-                    throw new RuntimeException('All relation records must be saved');
+                    throw new RuntimeException('All relation records must be saved. There are incorrect PKs.');
```

Comment:
You break existing logic. $savedRecords contains saved records count. $passedRecords contains setted in relation records count. Create another check condition with another exception text